### PR TITLE
Type `core.gui.custom_{widgets,dialogs}`

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
:exploding_head:

### CODE REVIEW

File: parsec/core/gui/custom_dialogs.py [hunk 1]
```diff
@@ -199,24 +224,39 @@ def on_finished(self):
         # is not set to None. Linux seems to clear them automatically over time.
         # Resetting the parent on MacOS causes a crash.
         if sys.platform != "darwin":
-            self.setParent(None)
+            # Mypy This is the correct way to remove a parent from a widget but PyQt miss that overload
+            self.setParent(None)  # type: ignore[call-overload]
+
+
+def exec_method(
```
Comment (vxgmichel) [inline parsec/core/gui/custom_dialogs.py:231]: Maybe call it `generate_dialog_method` or something like that?

File: parsec/_subprocess_dialog.py [hunk 2]
```diff
@@ -85,9 +86,17 @@ def load_resources(with_printer=False):
             QPrinter(QPrinter.HighResolution)
 
 
-def run_dialog(cls, method, *args, **kwargs):
+T_CLASS = TypeVar("T_CLASS")
+
+
+def run_dialog(
+    sub_cls: T_CLASS,
```
Comment (vxgmichel) [inline parsec/_subprocess_dialog.py:93]: `typing.Type` is most likely enough here.

File: parsec/core/gui/custom_dialogs.py [hunk 3]
```diff
@@ -53,9 +66,15 @@
     QEventLoop,
 )
 
+P = ParamSpec("P")
+R = TypeVar("R")
+T_CLASS = TypeVar("T_CLASS")
```
Comment (vxgmichel) [inline parsec/core/gui/custom_dialogs.py:71]: `typing.Type` is most likely enough here.

File: parsec/core/gui/custom_dialogs.py [hunk 4]
```diff
@@ -199,24 +224,39 @@ def on_finished(self):
...
 
-    def __init__(self, parent=None):
+    def __init__(self, parent: QWidget | None = None) -> None:
         super().__init__(QWidget(), "", parent, hide_close=True)
         self.MainWidget.hide()
 
     @classmethod
     @contextmanager
-    def manage_pools(cls):
+    def manage_pools(cls) -> Generator[None, None, None]:
```
Comment (vxgmichel) [inline parsec/core/gui/custom_dialogs.py:259]: Use `Iterator[None]`

**Review by vxgmichel — APPROVED** [review]
A couple of minor changes but LGTM otherwise :+1:

File: parsec/core/gui/custom_dialogs.py [hunk 5]
```diff
@@ -53,9 +66,15 @@
...
+P = ParamSpec("P")
+R = TypeVar("R")
+T_CLASS = TypeVar("T_CLASS")
+
 
 @contextmanager
-def bring_process_window_to_top(pool, timeout=3.0):
+def bring_process_window_to_top(
+    pool: multiprocessing.pool.Pool, timeout: float = 3.0
+) -> Generator[None, None, None]:
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply parsec/core/gui/custom_dialogs.py:77]: done

File: parsec/core/gui/custom_dialogs.py (hunk 1 above)
Comment (FirelightFlagboy) [reply parsec/core/gui/custom_dialogs.py:231]: done

File: parsec/_subprocess_dialog.py (hunk 2 above)
Comment (FirelightFlagboy) [reply parsec/_subprocess_dialog.py:93]: done

File: parsec/core/gui/custom_dialogs.py (hunk 3 above)
Comment (FirelightFlagboy) [reply parsec/core/gui/custom_dialogs.py:71]: Mypy complains about missing type parameters for generic type `Type`

File: parsec/core/gui/custom_dialogs.py (hunk 4 above)
Comment (FirelightFlagboy) [reply parsec/core/gui/custom_dialogs.py:259]: done

**Pull request merged**